compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1888,12 +1891,16 @@ impl CheckAttrVisitor<'_> {
 
         // Error on repr(transparent, <anything else>).
         if is_transparent && hints.len() > 1 {
-            let hint_spans: Vec<_> = hint_spans.clone().collect();
+            let hint_spans = hint_spans.clone().collect();
             self.tcx.sess.emit_err(errors::TransparentIncompatible {
                 hint_spans,
                 target: target.to_string(),
             });
         }
+        if is_explicit_rust && (int_reprs > 0 || is_c || is_simd) {
```

Comment:
I could also turn the bools into a bitfield and check if the difference between the set bitflags and {Rust, align, packed} is empty to address “but not necessarily limited to” part of

> We should ban all other repr(Rust, other) combinations, including, but not necesseraly limited to

Reply:
I think refactoring this function would be a lot of good, as the current code seems to be error-prone and confusing. But obv this refactoring can come as a follow-up, which does not need a backport. (cc me if you'll do it please)

Reply:
One of the things that would be nice is to _default to disallowing_, so that no changes can _accidentally_ allow new combinations.